src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```
@@ -572,4 +588,33 @@ public Response getApprovedUsers(@Auth DuosUser duosUser, @PathParam("identifier
       return createExceptionResponse(e);
     }
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  @Path("/{id}/nihInstitutionalCertification")
+  @RolesAllowed({ADMIN, DATASUBMITTER, CHAIRPERSON, MEMBER})
+  public Response getNihInstitutionalCertification(@Auth DuosUser duosUser, @PathParam("id") Integer id) {
+    User requestingUser = duosUser.getUser();
+    Dataset dataset = datasetService.findDatasetById(requestingUser, id);
+    if (dataset == null) {
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+    FileStorageObject nihFile = dataset.getNihInstitutionalCertificationFile();
+    if (nihFile != null &&
+        !nihFile.getDeleted() &&
+        nihFile.getFileName() != null &&
+        nihFile.getBlobId() != null &&
+        (requestingUser.hasUserRole(UserRoles.ADMIN)
+        || dataset.isCreator(requestingUser)
+        || dataset.isCustodian(requestingUser)
+        || requestingUser.verifyDACRole(CHAIRPERSON, dataset.getDacId())
+        || requestingUser.verifyDACRole(MEMBER, dataset.getDacId()))) {
+      InputStream fileStream = gcsService.getDocument(nihFile.getBlobId());
+      StreamingOutput streamOutput = createStreamingOutput(fileStream);
+      return Response.ok(streamOutput).header(HttpHeaders.CONTENT_DISPOSITION,
+          String.format("attachment; filename=\"%s\"", nihFile.getFileName())).build();
+    } else {
+      return Response.status(Status.NOT_FOUND).build();
```

Same feedback here, throw instead of return.